packages/repository/src/relations/belongs-to/belongs-to.decorator.ts

```diff
@@ -53,3 +58,22 @@ export function belongsTo<T extends Entity>(
     relation(meta)(decoratedTarget, decoratedKey);
   };
 }
+
+/**
+ * Sugar syntax for belongsTo decorator for hasOne relation. Calls belongsTo
+ * with property definition defaults for non-generated id field
+ * @param targetResolver A resolver function that returns the target model for
+ * a belongsTo relation
+ * @param definition Optional metadata for setting up a belongsTo relation
+ * @returns {(target: Object, key:string)}
+ */
+export function belongsToUniquely<T extends Entity>(
```

Comment:
The name is not easy to understand. The purpose is to say the FK should be unique.

Reply:
Maybe it's a warning sign that we should not overload `@belongTo` for responsibility of `@property`.

Reply:
I agree that ideally, relation decorators like `@belongsTo` and `@hasMany` should not call `@property` decorator under the hood. To get there, we would have to change the current design in a non-trivial way and break backwards compatibility along the way.

Do we want to delay `hasOne` implementation?

Personally, I'd prefer to finish `hasOne` implementation using the current approach, and then look into ways how to change the overall design.

> The name is not easy to understand. The purpose is to say the FK should be unique.

Can you propose a better name that's easier to understand?

Our intention here is to provide a syntactic sugar that will allow users to tell what they want to accomplish (setup the other side of hasOne relation) without going into low-level details (setting up a PK property as a way to get uniqueness constraint + make queries fast).